libraries/bezier-rs/src/subpath/solvers.rs

```diff
@@ -59,6 +63,34 @@ impl Subpath {
 
 		intersection_t_values
 	}
+
+	/// Returns a list of `t` values that correspond to the self intersection points of the subpath. For each intersection point, the returned `t` value is the smaller of the two that correspond to the point.
+	/// - `error` - For intersections with non-linear beziers, `error` defines the threshold for bounding boxes to be considered an intersection point.
+	/// - `minimum_seperation`: the minimum difference 2 adjacent `t`-values must have when comparing adjacent `t`-values in sorted order.
+	/// If the comparison condition is not satisfied, the function takes the larger `t`-value of the 2
+	///
+	///  **NOTE**: if an intersection were to occur within an `error` distance away from an anchor point, the algorithm will filter that intersection out.
+	pub fn self_intersections(&self, error: Option<f64>, minimum_seperation: Option<f64>) -> Vec<f64> {
+		let mut intersections_vec = Vec::new();
+		let n = self.len_segments();
+		// TODO: optimization opportunity - this for-loop currently compares all intersections with all curve-segments in the subpath collection
+		self.iter().enumerate().for_each(|(i, other)| {
+			intersections_vec.extend(other.self_intersections(error).iter().map(|value| (value[0] + (i as f64)) / (n as f64)));
+			self.iter().enumerate().skip(i + 1).for_each(|(j, curve)| {
+				intersections_vec.extend(
+					curve
+						.intersections(&other, error, minimum_seperation)
+						.iter()
+						.filter(|&value| {
+							let err = error.unwrap_or(MAX_ABSOLUTE_DIFFERENCE);
+							value > &err && (1. - value) > err
+						})
+						.map(|value| (value + (j as f64)) / (n as f64)),
```

Comment:
Could you simplify this to return tuples of `(curve_index, t_value)`? I haven't been able to make the ComputeType changes for subpaths, but I think within our own code local t values are so much easier to work with than the global t value. 

This will make it more annoying to draw the points in the wasm folder, but I've done the same thing in Project, so I think returning  `(curve_index, t_value)` is more consistent.

Reply:
Just to be clear, you want to change the return of `intersections` and `self_intersections` to be `Vec<(usize, f64)>`, correct?

Reply:
Yes please, thanks!